### DISCUSSION TITLE
Shutdown function with py-button

### DISCUSSION THREAD

**Discussion by KSuondProject** [pyscript/pyscript#1337] [Q&A] [answered]
![Schermata del 2023-03-30 16-18-54](https://user-images.githubusercontent.com/70426429/228866114-d447da00-2f77-4e2a-92f3-5e881bab8f3b.png)

Hi,
Can someone tell me if I can use something to shutdown my Raspberry?

Thank you in advance!

**Comment by JeffersGlass** [comment, marked as answer]
I don't think this is something that's possible in pure PyScript - the `os` module here (and really, all of the code) is running in your Browser, not on the hardware of the raspberry pi itself, and it only interacts with the browser's API and the virtual filesystem.

In order for this to be possible, web browsers would need to allow arbitrary websites to run down you computer, which they don't do for hopefully obvious reasons. Imagine going to Reddit and POOF, computer shuts down. Not ideal.

You may want to look at a framework like Flask or FastAPI to run some code on your Raspberry Pi; once you have code running there, it has much more power over what it does to your computer.

**Reply by KSuondProject** [reply]
Really thanks for the super fast answer.
So the best way is to run a webserver like Flask that listen the button status on a html page and not just use a stand alone html page.
I'll start learning flask :)

Thank you again!
